src/main/java/facades/StudentFacade.java

```diff
@@ -35,10 +35,13 @@ private EntityManager getEntityManager() {
     public Student getStudentByDatabaseId(long id) throws Exception {
         EntityManager em = getEntityManager();
         try {
+            if(em.find(Student.class, id) == null){
+                throw new IllegalArgumentException();
+            }
             Student student = em.find(Student.class, id);
```

Comment:
Wouldn't this mean you do the `em.find` operation twice?
I think a better approach is to do the operation, check the result, and then return based on that.

Reply:
Yes, you're right - that was a "didn't think, just wrote" part of the code :P Thanks!